test/test_error_lifecycle.rb

```diff
@@ -0,0 +1,93 @@
+require 'test/unit'
+require 'timeout'
+require 'thread'
+
+class TestTimeout < Test::Unit::TestCase
```

Comment:
`TestTimeout` is already used at https://github.com/ruby/timeout/blob/master/test/test_timeout.rb#L5. You should rename this.

Reply:
done

Reply:
got this failure in _only_ truffleruby on macos after that change 

https://github.com/jjb/timeout/actions/runs/5439839222/jobs/9892180910

it succeeded on a re-run: https://github.com/jjb/timeout/actions/runs/5439839222/jobs/9892337056

maybe it's a fluke caused by the CI environment

but maybe it's an indication of a race condition in the code?